### PR TITLE
Add apps icon list to machines table

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Add apps icon list to machines table

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/models
- Click on any model to view details page
- Click 'Machines' tab
- Verify that table now has 'apps' column showing apps icons

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1519

Screenshot 2021-01-12 at 12 39 49

### CODE REVIEW

File: src/pages/Models/Details/ModelDetails.test.js [hunk 1]
```diff
@@ -321,4 +321,26 @@ describe("ModelDetail Container", () => {
...
+            <TestRoute path="/models/*">
+              <ModelDetails />
+            </TestRoute>
+          </QueryParamProvider>
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const machineApps = wrapper.find(".machine-app-icons img");
+    expect(machineApps.length).toBe(10);
```
Comment (hatched) [inline src/pages/Models/Details/ModelDetails.test.js:344]: This test just checks that there are images on the page not that they are rendered appropriately per machine. Maybe spot check a few machines, the one with 4 images and one with 1?

File: src/pages/Models/Details/_model-details.scss [hunk 2]
```diff
@@ -226,6 +230,7 @@
   .entity-icon {
     border-radius: 50%;
     height: 1.5rem;
+    margin-bottom: 0.15rem;
```
Comment (hatched) [inline src/pages/Models/Details/_model-details.scss:233]: The icons are a bit too high in the row.

Screen Shot 2021-01-12 at 11 40 07 AM

File: src/pages/Models/Details/generators.js [hunk 3]
```diff
@@ -409,6 +412,27 @@ export function generateMachineRows(
...
+              appsOnMachine.push([appName, appInfo.charm]);
+            }
+          });
+      });
+
+    const apps = appsOnMachine.length
+      ? appsOnMachine.map((app) => {
+          return generateIconImg(app[0], app[1]);
+        })
+      : "There are no apps on this machine";
```
Comment (hatched) [inline src/pages/Models/Details/generators.js:432]: This is a bit too long and wraps, maybe just...
```suggestion
      : "none";
```
or
```suggestion
      : "";
```

Screen Shot 2021-01-12 at 11 51 10 AM

**Review by hatched — CHANGES_REQUESTED** [review]
Thanks @barrymcgee just some small QA issues.

File: src/pages/Models/Details/generators.js (hunk 3 above)
Comment (barrymcgee) [reply src/pages/Models/Details/generators.js:432]: That's not the design?

**Comment by long-chung** [comment]
"None" Works better

File: src/pages/Models/Details/_model-details.scss [hunk 4]
```diff
@@ -188,23 +188,27 @@
...
       }
 
       &:nth-child(3) {
-        width: 16%;
+        width: 10%;
       }
 
       &:nth-child(4) {
-        width: 25%;
+        width: 20%;
```
Comment (anthonydillon) [inline src/pages/Models/Details/_model-details.scss:203]: AZ doesnt need so much space. Maybe drop this to 15% and give instance ID (nth(5)) 15% too?

**Review by hatched — APPROVED** [review]

**Pull request merged**